### PR TITLE
d&i AssignSetOf[Objects/GeneratingMorphisms]

### PR DESCRIPTION
**Pull request by kamalsaleh**
to simplify working with objects and morphisms.

### CODE REVIEW

File: gap/Algebroids.gi [hunk 1]
```diff
@@ -57,7 +57,46 @@ InstallMethod( SetOfObjects,
...
+    ListN( names, objects, func );
+    
+end );
+
+##
+InstallOtherMethod( AssignSetOfObjects,
+        [ IsAlgebroid and HasUnderlyingQuiver ],
+  function( A )
+  
+    AssignSetOfObjects( A, "" );
```
Comment (mohamed-barakat) [inline gap/Algebroids.gi]: What does this line mean?

File: gap/Algebroids.gi [hunk 2]
```diff
@@ -66,6 +105,46 @@ InstallMethod( SetOfGeneratingMorphisms,
...
+    ListN( names, morphisms, func );
+    
+end );
+
+##
+InstallOtherMethod( AssignSetOfGeneratingMorphisms,
+        [ IsAlgebroid and HasUnderlyingQuiver ],
+  function( A )
+    
+    AssignSetOfGeneratingMorphisms( A, "" );
```
Comment (mohamed-barakat) [inline gap/Algebroids.gi]: What does this line mean?

File: gap/Algebroids.gi (hunk 1 above)
Comment (kamalsaleh) [reply gap/Algebroids.gi]: if the method is called with two arguments then concatenate the second argument with the labels of the vertices. If the method is called with one argument, i.e., with algebroid, then just use the labels of  the vertices. For example, if the algebroid is "a:1->2", then I would apply
`AssignSetOfObjects( A, "P_" );` But if the algebroid is `"a:T_1->T_2"`, then I would just apply 
`AssignSetOfObjects( A );` which calles `AssignSetOfObjects( A, "" );`

File: gap/Algebroids.gi (hunk 2 above)
Comment (kamalsaleh) [reply gap/Algebroids.gi]: Same as above, it gives the flexibility to use the names of the arrows or first attach some string to them.

**Pull request merged**